proxy/grpcproxy/watch.go

```diff
@@ -234,8 +237,35 @@ func (wps *watchProxyStream) sendLoop() {
 			if !ok {
 				return
 			}
-			if err := wps.stream.Send(wresp); err != nil {
-				return
+			// The grpc package's defaultServerMaxSendMessageSize, defaultServerMaxReceiveMessageSize
+			// and the maxSendMesgSize are set to lower numbers in order to demonstrate
+			// the fragmenting
+			maxEventsPerMsg := 20
+			watchRespSent := false
+			for !watchRespSent {
+				if maxEventsPerMsg == 0 {
+					return
+				}
+				for _, fragment := range v3rpc.FragmentWatchResponse(maxEventsPerMsg, wresp) {
```

Comment:
The watch client could have a `WithFragments()` option to skip reassembly. An alternative is to share the fragmenting code between the proxy and server, but it's not clear to me how that would fit together cleanly. Duplicating the fragmentation policy is not good-- too easy for code to drift.